kubectl-cloudflow/cmd/deploy.go

```diff
@@ -81,32 +105,195 @@ the stored credentials.
 	}
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.blueprintFilePath, "blueprint", "b", "", "blueprint file path.")
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.imagePath, "image", "i", "", "docker image path.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.username, "username", "u", "", "docker registry username.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.password, "password", "p", "", "docker registry password.")
 	deployOpts.cmd.Flags().BoolVarP(&deployOpts.passwordStdin, "password-stdin", "", false, "Take the password from stdin")
 	deployOpts.cmd.Flags().StringArrayVar(&deployOpts.volumeMounts, "volume-mount", []string{}, "Accepts a key/value pair separated by an equal sign. The key should be the name of the volume mount, specified as '[streamlet-name].[volume-mount-name]'. The value should be the name of an existing persistent volume claim.")
+	deployOpts.cmd.Flags().StringToIntVar(&deployOpts.replicas, "scale", map[string]int{}, "Accepts key/value pairs for replicas per streamlet")
 
 	rootCmd.AddCommand(deployOpts.cmd)
 }
 
 func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 	version.FailOnProtocolVersionMismatch()
 
-	imageRef := args[0]
+	if cmd.Flag("image").Changed == true {
+		deployImage(cmd, opts, args)
+	} else {
+		deployBlueprint(cmd, opts, args)
+	}
+}
+
+func deployBlueprint(cmd *cobra.Command, opts *deployOptions, args []string) {
+	// read blueprint file and load its content
+	blueprintFile := cmd.Flag("blueprint").Value.String()
+	content, err := ioutil.ReadFile(blueprintFile)
+	if err != nil {
+		util.LogAndExit("Cannot read blueprint file: Error : %s", err.Error())
+	}
+	// this needs to be replaced with calls from blueprint package once verify blueprint is in
+	blueprintConfig, err := loadBlueprintConfigContent(string(content[:]))
+	if err != nil {
+		util.LogAndExit("Cannot load blueprint file information: Error : %s", err.Error())
+	}
+
+	// get the images from the blueprint
+	imagesByName := blueprintConfig.GetNode("blueprint.images").GetObject().Items()
+
+	var imageRefs []deploy.ImageReference
+	var dockerRegistryURL string
+
+	for _, imageRef := range imagesByName {
+		ref, err := parseImageReference(imageRef.GetString())
+		if err != nil {
+			util.LogAndExit("Cannot parse image reference from blueprint: Error : %s", err.Error())
+		}
+		imageRefs = append(imageRefs, *ref)
+
+		// currently we assume all images in the blueprint will have the same registry
+		// hence overwriting is ok. However we need to handle this differently when we have 
+		// support for multiple registries in blueprint
+		dockerRegistryURL = ref.Registry
+	}
+
+	applicationSpec, pulledImages := deploy.CreateApplicationSpecFromBlueprintAndImages(imageRefs, blueprintConfig, opts.replicas)
+
+	namespace := applicationSpec.AppID
+	k8sClient, k8sErr := k8s.GetClient()
+	if k8sErr != nil {
+		util.LogAndExit("Failed to create new kubernetes client for Cloudflow application `%s`, %s", namespace, k8sErr.Error())
```

Comment:
Is this the same as could not connect? I think this text can be improved, to be more user friendly, less implementation jargon:

'Failed to connect to kubernetes cluster: %s'
I think reporting the name of the app back is not necessary (for Cloudflow application `%s`), they just typed it in, and they know this is cloudflow...

Reply:
done ..